### ISSUE TITLE
problem after server migration

### ISSUE THREAD

**Issue by FreeBeeEssDee** [coppermine-gallery/cpg1.6.x#65]
My coppermine was migrated from an old server with freebsd 12 and php 7.4 to a new machine with freebsd 13 stable and php 8.1. On the new server all other web applications (wordpress, nextcloud, postfixadmin, ...) are working smoothly. But coppermine throws the error:
Fatal error: Uncaught Error: mysqli object is already closed in /usr/local/www/coppermine/include/database/mysqli/dbase.inc.php:73 Stack trace: #0 /usr/local/www/coppermine/include/init.inc.php(172): CPG_Dbase->getError() #1 /usr/local/www/coppermine.3/index.php(22): require('...') #2 {main} thrown in /usr/local/www/coppermine/include/database/mysqli/dbase.inc.php on line 73

I tried the same coppermine from the old server, a freshly installed package from the ports and 1.6.20 and the development version from here. All with the same result. I successfully checked the database credentials by using the mariadb command-line tool.

**Comment by ron4mac** [comment]
It is likely the value for the database server that is causing an issue for you. Try using just 'localhost' for the database server.

**Comment by FreeBeeEssDee** [comment]
Thanks for the hint and my fault that I forgot to include my config.inc.php:
cat config.inc.php
<?php
// Coppermine configuration file
// Database configuration
$CONFIG['dbtype'] =   'mysqli';			// Your database type
$CONFIG['dbserver'] = 'localhost';			// Your database server
$CONFIG['dbuser'] =   'username';			// Your database username
$CONFIG['dbpass'] =   'dbpassword';			// Your database password
$CONFIG['dbname'] =   'dbname';			// Your database name
// DATABASE TABLE NAMES PREFIX
$CONFIG['TABLE_PREFIX'] = 'cpg15x_';

I already use localhost and the database can be reached:

telnet localhost 3306
Trying ::1...
telnet: connect to address ::1: Connection refused
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Y
5.5.5-10.6.10-MariaDBVH	0>%<?gi[�l-9udt=n6]Vtmysql_native_password

Therefore I doubt that the basic database connectivity is the culprit.
What I also forgot to mention: php is now used as php-fpm. But I double checked that default_host and default_socket are set correctly in php.ini

**Comment by ron4mac** [comment]
resolved?

**Comment by FreeBeeEssDee** [comment]
No, I already used localhost and I have checked with "telnet localhost 3306" that the database can be reached.

**Comment by ron4mac** [comment]
Run the php command 'phpinfo();' and look at the mysqli section to see that it is configured properly.
Or try 'localhost:3306' as the database server in config.inc.php

**Comment by ron4mac** [comment]
In the file _include/init.inc.php_, on lines 172 and 173, change the 2 **$CPGDB->getError()** calls to **$CPGDB->getError(false,true)** to see if that gives you any better information.

**Comment by ron4mac** [comment]
Has this issue been resolved?
